crates/task-impls/src/quorum_proposal.rs

```diff
@@ -0,0 +1,541 @@
+                            event.clone(),
+                        );
+                    }
+                    either::Left(qc) => {
+                        let mut consensus = self.consensus.write().await;
+                        consensus.high_qc = qc.clone();
+
+                        // cancel poll for votes
+                        self.quorum_network
+                            .inject_consensus_info(ConsensusIntentEvent::CancelPollForVotes(
+                                *qc.view_number,
+                            ))
+                            .await;
+
+                        // We need to drop our handle here to make the borrow checker happy.
+                        drop(consensus);
+                        debug!(
+                            "Attempting to publish proposal after forming a QC for view {}",
+                            *qc.view_number
+                        );
+
+                        let view = qc.view_number + 1;
+
+                        self.create_dependency_task_if_new(
+                            view,
+                            event_receiver,
+                            event_sender,
+                            event.clone(),
+                        );
+                    }
+                }
+            }
+            HotShotEvent::SendPayloadCommitmentAndMetadata(payload_commitment, _metadata, view) => {
+                let view = *view;
+                if view < self.latest_proposed_view {
+                    debug!(
+                        "Payload commitment is from an older view {:?}",
+                        view.clone()
+                    );
+                    return;
+                }
+
+                debug!("Got payload commitment and meta {:?}", payload_commitment);
+
+                self.create_dependency_task_if_new(
+                    view,
+                    event_receiver,
+                    event_sender,
+                    event.clone(),
+                );
+            }
+            HotShotEvent::ViewSyncFinalizeCertificate2Recv(view_sync_finalize_cert) => {
+                let view = view_sync_finalize_cert.view_number;
+                if view < self.latest_proposed_view {
+                    debug!(
+                        "View sync certificate is from an old view {:?}",
+                        view.clone()
+                    );
+                    return;
+                }
+
+                if !validate_view_sync_finalize_certificate(
+                    view_sync_finalize_cert.clone(),
+                    event_sender.clone(),
+                ) {
+                    return;
+                }
+
+                self.create_dependency_task_if_new(view, event_receiver, event_sender, event);
+            }
+            HotShotEvent::QuorumProposalRecv(proposal, _sender) => {
+                let view = proposal.data.get_view_number();
+                if view < self.latest_proposed_view {
```

Comment:
We should also skip the view that has been proposed, so this should be `<=`.

Reply:
Fixed this in my branch by making view be `view + 1` as this was a mistake to not add one to it before.